File: examples/docs/modules/ROOT/examples/akkastreams-scala/src/main/scala/cloudflow/akkastreamsdoc/DataStreamingIngress.scala
package cloudflow.akkastreamsdoc

import akka.http.scaladsl.marshallers.sprayjson.SprayJsonSupport._
import akka.http.scaladsl.common.EntityStreamingSupport

import cloudflow.akkastream._
import cloudflow.akkastream.util.scaladsl._

import cloudflow.streamlets.avro._
import cloudflow.streamlets._

import JsonSupport._

class DataStreamingIngress extends AkkaServerStreamlet {
  val out   = AvroOutlet[Data]("out", RoundRobinPartitioner)
  def shape = StreamletShape.withOutlets(out)

  implicit val entityStreamingSupport = EntityStreamingSupport.json()
  override def createLogic            = HttpServerLogic.defaultStreaming(this, out)
}

File: examples/snippets/modules/ROOT/examples/akkastreams-java/src/main/java/cloudflow/akkastreamsdoc/DataStreamingIngress.java
package cloudflow.akkastreamsdoc;

// tag::httpStreamingIngress[]
import akka.http.javadsl.common.EntityStreamingSupport;
import akka.http.javadsl.marshallers.jackson.Jackson;

import cloudflow.akkastream.AkkaServerStreamlet;

import cloudflow.akkastream.AkkaStreamletLogic;
import cloudflow.akkastream.util.javadsl.HttpWriterLogic;
import cloudflow.streamlets.RoundRobinPartitioner;
import cloudflow.streamlets.StreamletShape;
import cloudflow.streamlets.avro.AvroOutlet;

public class DataStreamingIngress extends AkkaServerStreamlet {

  private AvroOutlet<Data> out =
      AvroOutlet.create("out", Data.class)
          .withPartitioner(RoundRobinPartitioner.getInstance());

  public StreamletShape shape() {
    return StreamletShape.createWithOutlets(out);
  }

  public AkkaStreamletLogic createLogic() {
    EntityStreamingSupport ess = EntityStreamingSupport.json();
    return HttpWriterLogic.createDefaultStreaming(
        this, out, Jackson.byteStringUnmarshaller(Data.class), ess, getContext());
  }
}
// end::httpStreamingIngress[]